File: RxTools-library/src/main/java/com/vondear/rxtools/view/progressing/sprite/CircleSprite.java
package com.vondear.rxtools.view.progressing.sprite;

import android.animation.ValueAnimator;
import android.graphics.Canvas;
import android.graphics.Paint;

/**
 * Created by ybq.
 */
public class CircleSprite extends ShapeSprite {

    @Override
    public ValueAnimator onCreateAnimation() {
        return null;
    }

    @Override
    public void drawShape(Canvas canvas, Paint paint) {
        if (getDrawBounds() != null) {
            int radius = Math.min(getDrawBounds().width(), getDrawBounds().height()) / 2;
            canvas.drawCircle(getDrawBounds().centerX(),
                    getDrawBounds().centerY(),
                    radius, paint);
        }
    }
}

File: RxUI/src/main/java/com/tamsiree/rxui/view/loadingview/sprite/CircleSprite.kt
package com.tamsiree.rxui.view.loadingview.sprite

import android.animation.ValueAnimator
import android.graphics.Canvas
import android.graphics.Paint

/**
 * @author tamsiree
 */
open class CircleSprite : ShapeSprite() {
    override fun onCreateAnimation(): ValueAnimator? {
        return null
    }

    override fun drawShape(canvas: Canvas?, paint: Paint?) {
        if (drawBounds != null) {
            val radius = Math.min(drawBounds.width(), drawBounds.height()) / 2
            canvas!!.drawCircle(drawBounds.centerX().toFloat(),
                    drawBounds.centerY().toFloat(),
                    radius.toFloat(), paint!!)
        }
    }
}